File: task1/task1.cpp
#include <iostream>
#include <iomanip>
#include <cmath>

using namespace std;

double f(double x)
{
    if (x < -4)
    {
        // Левая ветка графика
        return -sqrt(-x - 4) - 1;
    }
    else if (x <= 4)
    {
        // Средняя часть графика
        return -sin(M_PI * x / 4.0);
    }
    else
    {
        // Правая ветка графика
        return sqrt(x - 4) + 1;
    }
}

int main()
{
    double xStart, xEnd, dx;

    cout << "Введите начало интервала: ";
    cin >> xStart;

    cout << "Введите конец интервала: ";
    cin >> xEnd;

    cout << "Введите шаг: ";
    cin >> dx;

    cout << fixed << setprecision(3);

    cout << "\n---------------------\n";
    cout << "|    x    |    y    |\n";
    cout << "---------------------\n";

    for (double x = xStart; x <= xEnd; x += dx)
    {
        cout << "| "
             << setw(7) << x
             << " | "
             << setw(7) << f(x)
             << " |\n";
    }

    cout << "---------------------\n";

    return 0;
}

File: task1/task1.go
package main

import (
	"fmt"
	"math"
)

func f(x float64) float64 {

	if x < -4 {
		// Левая ветка
		return -math.Sqrt(-x - 4)
	} else if x <= 4 {
		// Средняя часть
		return -math.Sin(math.Pi * x / 4.0)
	} else {
		// Правая ветка
		return math.Sqrt(x-4) + 1
	}
}

func main() {

	xStart := -6.0
	xEnd := 6.0
	dx := 0.5

	fmt.Println("---------------------")
	fmt.Println("|    x    |    y    |")
	fmt.Println("---------------------")

	for x := xStart; x <= xEnd; x += dx {

		fmt.Printf("| %7.3f | %7.3f |\n",
			x,
			f(x))
	}

	fmt.Println("---------------------")
}
